src/main/java/tr/edu/ogu/ceng/controller/StudentController.java

```diff
@@ -41,14 +41,15 @@ public Student addStudent(@RequestBody Student student) {
 		return studentService.addStudent(student);
 	}
 
-	@PutMapping()
-	public ResponseEntity<Student> updateStudent(@RequestBody Student student) {
-		return ResponseEntity.ok(student);
+	@PutMapping("/{id}")
+	public Student updateStudent(@RequestBody Student student, @PathVariable(name = "id") long id) {
```

Comment:
neden id ayrı bir parametre olarak geliyor?

Reply:
Hocam bu metotu ben yazmadığım için nedenini bilmiyorum.

Reply:
Ama bu branch ve PR sana aitse değişiklik görüyorum yani bu değişikliği sen yapmış görünüyorsun.

Reply:
Hocam  conflict çözmek için git pull yaptığımda bu değişiklikleri ben yapmışım gibi gözüküyor ama tam nedenini bilmiyorum.Daha öncede aynı sorun olmuştu nasıl çözebilirim?

Reply:
pull ile çektiğin bir dosyada sen bilerek ya da bilmeyerek herhangi bir değişiklik yapmadıysan değişiklik görünmesi imkansız. 
Nedeni basit; pull ile main'deki dosyayı çekiyorsun. PR çıkarken karşılaştırılan dosya senin branchindeki ile main branchindeki. Şimdi main'den çektiğin dosya ile main'deki dosya farklı olabilir mi? Hayır olamaz. Ama sen bir değişiklik yaparsan fark oluşur.  

Eklenen bir kaç boşluk formatlamadan kaynaklanabilir, bunu görmezden gelebiliriz ancak burada main branch'inde olmayan bir id değişkeni ekliyorsun. 

Şimdi bunu sen yazmamış olabilirsin ki zaten öyle ama git pull vs kullanırken dikkat etmediğiniz birşeyler var. Bu yüzden main branch'i ile senkron olduğunuzu teyit edin gerekirse git fetch ile güncelleyin ve git restore ile değişiklik yapmadığınız dosyalardaki değişiklikleri iptal edin. kodlarınızı göndermeden önce git status ile bakıp student dosyasında ben bir değişiklik yapmadık ki neden değişiklik görünüyor sorusunu da sormalısınız.